salt/grains/metadata.py

```diff
@@ -70,11 +74,14 @@ def _search(prefix="latest/"):
             retdata = http.query(os.path.join(HOST, prefix, line)).get('body', None)
             # (gtmanfred) This try except block is slightly faster than
             # checking if the string starts with a curly brace
-            try:
-                ret[line] = salt.utils.json.loads(retdata)
-            except ValueError:
+            if isinstance(retdata, six.binary_type):
```

Comment:
I am not sure if the intent is for it to be bytes, but as it is written, I am getting bytes back from querying the api, and I only want to do a json.loads if it is json, otherwise it could be None.

Reply:
OK, I didn't know what the non-bytes value could be, so thanks for clearing that up. There's something else here though that might break on Python 3. It seems that `json.loads()` will load a bytestring in Python 3.6, but not 3.5:

### Python 3.6
```
Python 3.6.4 (default, Jan  5 2018, 02:35:40)
[GCC 7.2.1 20171224] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> import json
>>> json.loads(b'{"foo": "bar"}')
{'foo': 'bar'}
>>>
```

### Python 3.5
```
Python 3.5.4 (default, Aug 23 2017, 18:32:05) 
[GCC 6.4.1 20170727 (Red Hat 6.4.1-1)] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> import json
>>> json.loads(b'{"foo": "bar"}')
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/usr/lib64/python3.5/json/__init__.py", line 312, in loads
    s.__class__.__name__))
TypeError: the JSON object must be str, not 'bytes'
>>> 
```

So you will likely need to run to_str/to_unicode on retdata before trying to load it as JSON to ensure that Python < 3.6 doesn't raise an exception.

Reply:
:+1: